src/test/java/com/artipie/rpm/pkg/ModifiableMetadataTest.java

```diff
@@ -0,0 +1,136 @@
+import org.cactoos.list.ListOf;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.core.IsNot;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import org.redline_rpm.ReadableChannelWrapper;
+import org.redline_rpm.Scanner;
+import org.redline_rpm.header.Format;
+import org.redline_rpm.header.Header;
+
+/**
+ * Test for {@link ModifiableMetadata}.
+ * @since 0.9
+ * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
+ */
+class ModifiableMetadataTest {
+
+    @Test
+    void generatesMetadataFile(@TempDir final Path temp) throws IOException {
+        final Path res = temp.resolve("primary.xml");
+        res.toFile().createNewFile();
+        final ModifiableMetadata mtd = new ModifiableMetadata(
+            new MetadataFile("primary", new PrimaryOutput(res).start()),
+            Paths.get("src/test/resources-binary/repodata/primary.xml.example")
+        );
+        final Path rpm =
+            Paths.get("src/test/resources-binary/abc-1.01-26.git20200127.fc32.ppc64le.rpm");
+        mtd.accept(new FilePackage.Headers(this.header(rpm), rpm, Digest.SHA256));
+        mtd.close();
+        mtd.brush(
+            new ListOf<String>("7eaefd1cb4f9740558da7f12f9cb5a6141a47f5d064a98d46c29959869af1a44")
+        );
+        MatcherAssert.assertThat(
+            "Has 'abc' and 'nginx' packages",
+            new String(Files.readAllBytes(res), StandardCharsets.UTF_8),
+            XhtmlMatchers.hasXPath(
+                //@checkstyle LineLengthCheck (2 lines)
+                "/*[local-name()='metadata']/*[local-name()='package']/*[local-name()='name' and text()='abc']",
+                "/*[local-name()='metadata']/*[local-name()='package']/*[local-name()='name' and text()='nginx']"
+            )
+        );
+        MatcherAssert.assertThat(
+            "Does not have 'aom' package",
+            new String(Files.readAllBytes(res), StandardCharsets.UTF_8),
+            new IsNot<>(
+                XhtmlMatchers.hasXPath(
+                    //@checkstyle LineLengthCheck (1 line)
+                    "/*[local-name()='metadata']/*[local-name()='package']/*[local-name()='name' and text()='aom']"
+                )
+            )
+        );
+    }
+
+    @Test
+    void savesItselfToRepomd(@TempDir final Path temp) throws IOException, XMLStreamException {
+        final Path primary = temp.resolve("test.filelists.xml");
+        primary.toFile().createNewFile();
+        final ModifiableMetadata mtd = new ModifiableMetadata(
+            new MetadataFile("filelists", new FilelistsOutput(primary).start()),
+            Paths.get("src/test/resources-binary/repodata/filelists.xml.example")
+        );
+        mtd.close();
+        final Path repomd = temp.resolve("repomd.xml");
+        final XmlRepomd xml = new XmlRepomd(repomd);
+        xml.begin(System.currentTimeMillis() / Tv.THOUSAND);
+        mtd.save(StandardNamingPolicy.PLAIN, Digest.SHA256, xml);
+        xml.close();
```

Comment:
@olenagerasimova we can put `XmlRepoMd` in a try-with-resources block instead of invoking `close()`, it's a `Closeable`.